src/link.ts

```diff
@@ -298,32 +298,39 @@ export class Link implements esr.AbiProvider {
     ): Promise<TransactResult> {
         const t = transport || this.transport
         const broadcast = options ? options.broadcast !== false : true
-        // eosjs transact compat: upgrade to transaction if args have any header fields
-        let anyArgs = args as any
-        if (
-            args.actions &&
-            (anyArgs.expiration ||
-                anyArgs.ref_block_num ||
-                anyArgs.ref_block_prefix ||
-                anyArgs.max_net_usage_words ||
-                anyArgs.max_cpu_usage_ms ||
-                anyArgs.delay_sec)
-        ) {
-            args = {
-                transaction: {
-                    expiration: '1970-01-01T00:00:00',
-                    ref_block_num: 0,
-                    ref_block_prefix: 0,
-                    max_net_usage_words: 0,
-                    max_cpu_usage_ms: 0,
-                    delay_sec: 0,
-                    ...anyArgs,
-                },
+        try {
+            // eosjs transact compat: upgrade to transaction if args have any header fields
+            let anyArgs = args as any
+            if (
+                args.actions &&
+                (anyArgs.expiration ||
+                    anyArgs.ref_block_num ||
+                    anyArgs.ref_block_prefix ||
+                    anyArgs.max_net_usage_words ||
+                    anyArgs.max_cpu_usage_ms ||
+                    anyArgs.delay_sec)
+            ) {
+                args = {
+                    transaction: {
+                        expiration: '1970-01-01T00:00:00',
+                        ref_block_num: 0,
+                        ref_block_prefix: 0,
+                        max_net_usage_words: 0,
+                        max_cpu_usage_ms: 0,
+                        delay_sec: 0,
+                        ...anyArgs,
+                    },
+                }
             }
+            const request = await this.createRequest(args, t)
+            const result = await this.sendRequest(request, t, broadcast)
+            return result
+        } catch (error) {
+            if (t.onFailure) {
+                t.onFailure(undefined, error)
```

Comment:
This will cause onFailure to be called multiple times with different arguments for the same error